### ISSUE TITLE
Error retrieving calendar events with Outlook/Thunderbird

### ISSUE THREAD

**Issue by antonioborondo** [mguessan/davmail#328]
I get the following error when retrieving the calendar events with an Outlook account and Thunderbird:

```
2024-01-23 19:48:28,608 INFO  [CaldavConnection-53332] davmail.connection  - LOGON - 127.0.0.1:53332 fakemail@outlook.com
2024-01-23 19:48:28,610 DEBUG [CaldavConnection-53332] davmail  - > HTTP/1.1 207 Multi-Status
2024-01-23 19:48:28,610 DEBUG [CaldavConnection-53332] davmail  - > Server: DavMail Gateway 6.0.1-trunk
2024-01-23 19:48:28,610 DEBUG [CaldavConnection-53332] davmail  - > DAV: 1, calendar-access, calendar-auto-schedule, calendarserver-private-events, addressbook
2024-01-23 19:48:28,611 DEBUG [CaldavConnection-53332] davmail  - > Date: Tue, 23 Jan 2024 18:48:28 +0000
2024-01-23 19:48:28,611 DEBUG [CaldavConnection-53332] davmail  - > Expires: Tue, 23 Jan 2024 18:48:28 +0000
2024-01-23 19:48:28,611 DEBUG [CaldavConnection-53332] davmail  - > Cache-Control: private, max-age=0
2024-01-23 19:48:28,611 DEBUG [CaldavConnection-53332] davmail  - > Transfer-Encoding: chunked
2024-01-23 19:48:28,611 DEBUG [CaldavConnection-53332] davmail  - > Content-Type: text/xml;charset=UTF-8
2024-01-23 19:48:28,611 DEBUG [CaldavConnection-53332] davmail  - > Connection: keep-alive
2024-01-23 19:48:28,611 DEBUG [CaldavConnection-53332] davmail  - > 
2024-01-23 19:48:28,611 DEBUG [CaldavConnection-53332] davmail  - Report item 1/100
2024-01-23 19:48:28,861 DEBUG [CaldavConnection-53332] davmail.http.HttpClientAdapter  - Add cookie [version: 0][name: exchangecookie][value: 7c6de2e580ed4b90ab736ea8688cea01][domain: outlook.office365.com][path: /][expiry: null]
2024-01-23 19:48:28,862 DEBUG [CaldavConnection-53332] davmail.http.HttpClientAdapter  - Add cookie [version: 0][name: exchangecookie][value: 7c6de2e580ed4b90ab736ea8688cea01][domain: outlook.office365.com][path: /][expiry: null]
2024-01-23 19:48:28,862 DEBUG [CaldavConnection-53332] davmail.http.HttpClientAdapter  - Add cookie [version: 0][name: EXOBasicAuth][value: compactTicket=&key=fakemail%40outlook.com+microsoft.exchange.webservices+85.57.130.80+&signature=&puid=00030000AA7F40E5&expireTime=638416569870000000&membername=fakemail%40outlook.com&flags=False&userType=OutlookCom&aliasVersion=39&authInstant=1706031387&issueInstant=1706031387&recoveryInstant=0&clientIp=255.255.255.255&cid=f6c78572ac367566&ageGroup=3&rpsTicket=GABWAgMAAAAMgAAAEwEgunSouPd98XNcBuvnsw2HtUMpAsUL1A7S8jjVSi2LiG4AAcqZdiNXZ+2w6mgC4T75+eYIfOg8gM9YNvdWYknCgDaZUZzaDFhW1FM98HiD+7fCA/sEbv9948ETNQwCtqL8W0rS4ziK6nBijxc35DTZjc1ONR0zStg6UqzEaY4PXr5M2b6i+twBew7urkeprSPIz9N2W3sJ5GDjfORFiFV3ErNWKaXFsxemFBl/wLKMoO0vJdfTz3XeO1omCnoEKKkVAeLSg88veKChPL+RTiiWsZTLCn6uzQXdEy2vgWQyOFfDhQJ+5pdTlh9EZXe+IePmHunxOZqIvezaBDttwjYJXBRnBVFA26jj+/Ch4gCZKgg42EiryEq4Yw/adlrrkreay9IjAX4AIwEAAAMA5UB/qhv5r2Ub+a9l6XcEAAoRIIAAGwBhbnRvbmlvYm9yb25kb0BvdXRsb29rLmNvbQBjAAAnYW50b25pb2Jvcm9uZG8lb3V0bG9vay5jb21AcGFzc3BvcnQuY29tAAACMUVTAAAAmklFCAkCAAB3uVVAAAZDAAdBbnRvbmlvAAdCb3JvbmRvAAAAAAAAAAAAAAAAAAAAAAAArDZ1ZvbHhXIAABv5r2UboCZmAAAAAAAAAAAAAAAADQA4NS41Ny4xMzAuODAABScAAAAAAAAAAAAAAAABBAAAAAAAAAAAAAAAAAAAAMRyS6KgMDrcAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwAAAA==][domain: outlook.office365.com][path: /ews][expiry: Wed Jan 24 02:36:27 CET 2024]
2024-01-23 19:48:29,001 DEBUG [CaldavConnection-53332] davmail.http.HttpClientAdapter  - Add cookie [version: 0][name: exchangecookie][value: 7c6de2e580ed4b90ab736ea8688cea01][domain: outlook.office365.com][path: /][expiry: null]
2024-01-23 19:48:29,001 DEBUG [CaldavConnection-53332] davmail.http.HttpClientAdapter  - Add cookie [version: 0][name: exchangecookie][value: 7c6de2e580ed4b90ab736ea8688cea01][domain: outlook.office365.com][path: /][expiry: null]
2024-01-23 19:48:29,002 DEBUG [CaldavConnection-53332] davmail.http.HttpClientAdapter  - Add cookie [version: 0][name: EXOBasicAuth][value: compactTicket=&key=fakemail%40outlook.com+microsoft.exchange.webservices+85.57.130.80+&signature=&puid=00030000AA7F40E5&expireTime=638416569870000000&membername=fakemail%40outlook.com&flags=False&userType=OutlookCom&aliasVersion=39&authInstant=1706031387&issueInstant=1706031387&recoveryInstant=0&clientIp=255.255.255.255&cid=f6c78572ac367566&ageGroup=3&rpsTicket=GABWAgMAAAAMgAAAEwEgunSouPd98XNcBuvnsw2HtUMpAsUL1A7S8jjVSi2LiG4AAcqZdiNXZ+2w6mgC4T75+eYIfOg8gM9YNvdWYknCgDaZUZzaDFhW1FM98HiD+7fCA/sEbv9948ETNQwCtqL8W0rS4ziK6nBijxc35DTZjc1ONR0zStg6UqzEaY4PXr5M2b6i+twBew7urkeprSPIz9N2W3sJ5GDjfORFiFV3ErNWKaXFsxemFBl/wLKMoO0vJdfTz3XeO1omCnoEKKkVAeLSg88veKChPL+RTiiWsZTLCn6uzQXdEy2vgWQyOFfDhQJ+5pdTlh9EZXe+IePmHunxOZqIvezaBDttwjYJXBRnBVFA26jj+/Ch4gCZKgg42EiryEq4Yw/adlrrkreay9IjAX4AIwEAAAMA5UB/qhv5r2Ub+a9l6XcEAAoRIIAAGwBhbnRvbmlvYm9yb25kb0BvdXRsb29rLmNvbQBjAAAnYW50b25pb2Jvcm9uZG8lb3V0bG9vay5jb21AcGFzc3BvcnQuY29tAAACMUVTAAAAmklFCAkCAAB3uVVAAAZDAAdBbnRvbmlvAAdCb3JvbmRvAAAAAAAAAAAAAAAAAAAAAAAArDZ1ZvbHhXIAABv5r2UboCZmAAAAAAAAAAAAAAAADQA4NS41Ny4xMzAuODAABScAAAAAAAAAAAAAAAABBAAAAAAAAAAAAAAAAAAAAMRyS6KgMDrcAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwAAAA==][domain: outlook.office365.com][path: /ews][expiry: Wed Jan 24 02:36:27 CET 2024]
2024-01-23 19:48:29,004 DEBUG [CaldavConnection-53332] davmail.caldav.CaldavConnection  - status code: 404, reason phrase: AQMkADAwATMwMAItYWEAN2YtNDBlNS0wMAItMDAKAEYAAAPodCnhbNvQSqr8qr0ouP_IBwCRzmCgOF51Q4c0gPYjK3n2AAACAQ0AAACRzmCgOF51Q4c0gPYjK3n2AAYrWcODAAAA.EML not found in /users/fakemail@outlook.com/calendar
davmail.exception.HttpNotFoundException: status code: 404, reason phrase: AQMkADAwATMwMAItYWEAN2YtNDBlNS0wMAItMDAKAEYAAAPodCnhbNvQSqr8qr0ouP_IBwCRzmCgOF51Q4c0gPYjK3n2AAACAQ0AAACRzmCgOF51Q4c0gPYjK3n2AAYrWcODAAAA.EML not found in /users/fakemail@outlook.com/calendar
	at davmail.exchange.ews.EwsExchangeSession.getItem(EwsExchangeSession.java:2411)
	at davmail.caldav.CaldavConnection.reportItems(CaldavConnection.java:799)
	at davmail.caldav.CaldavConnection.handleFolderOrItem(CaldavConnection.java:294)
	at davmail.caldav.CaldavConnection.handleRequest(CaldavConnection.java:237)
	at davmail.caldav.CaldavConnection.run(CaldavConnection.java:180)
2024-01-23 19:48:29,004 WARN  [CaldavConnection-53332] davmail  - Item AQMkADAwATMwMAItYWEAN2YtNDBlNS0wMAItMDAKAEYAAAPodCnhbNvQSqr8qr0ouP_IBwCRzmCgOF51Q4c0gPYjK3n2AAACAQ0AAACRzmCgOF51Q4c0gPYjK3n2AAYrWcODAAAA.EML not available: /users/fakemail@outlook.com/calendar/AQMkADAwATMwMAItYWEAN2YtNDBlNS0wMAItMDAKAEYAAAPodCnhbNvQSqr8qr0ouP_IBwCRzmCgOF51Q4c0gPYjK3n2AAACAQ0AAACRzmCgOF51Q4c0gPYjK3n2AAYrWcODAAAA.EML
```
I'm using DavMail Gateway 6.0.1-trunk and this is the configuration:
```
davmail.ssl.keystoreType=
davmail.ssl.keystorePass=
davmail.proxyPassword=
davmail.smtpPort=1025
davmail.enableKerberos=false
davmail.folderSizeLimit=
davmail.forceActiveSyncUpdate=false
davmail.imapAutoExpunge=true
davmail.useSystemProxies=false
davmail.proxyUser=
davmail.ssl.nosecuresmtp=false
davmail.caldavPastDelay=0
davmail.ssl.keyPass=
log4j.logger.httpclient.wire=WARN
davmail.noProxyFor=
davmail.server=true
davmail.popMarkReadOnRetr=false
davmail.ssl.nosecureimap=false
davmail.disableTrayActivitySwitch=false
davmail.caldavAutoSchedule=true
davmail.enableProxy=false
davmail.proxyPort=
davmail.mode=EWS
davmail.smtpSaveInSent=true
davmail.bindAddress=
davmail.ssl.nosecurepop=false
davmail.ssl.pkcs11Library=
log4j.rootLogger=WARN
davmail.ssl.keystoreFile=
log4j.logger.davmail=DEBUG
davmail.ssl.clientKeystoreType=
davmail.clientSoTimeout=
davmail.ssl.pkcs11Config=
davmail.imapPort=1143
davmail.ssl.clientKeystorePass=
davmail.url=https://outlook.office365.com/EWS/Exchange.asmx
davmail.sentKeepDelay=0
davmail.ssl.nosecureldap=false
davmail.imapAlwaysApproxMsgSize=false
davmail.ssl.nosecurecaldav=false
davmail.popPort=1110
davmail.showStartupBanner=true
log4j.logger.httpclient=WARN
davmail.proxyHost=
davmail.ldapPort=1389
davmail.server.certificate.hash=
davmail.disableGuiNotifications=false
davmail.imapIdleDelay=
davmail.allowRemote=false
davmail.disableUpdateCheck=false
davmail.caldavPort=1080
davmail.enableKeepAlive=false
davmail.ssl.clientKeystoreFile=
davmail.logFilePath=.local/state/davmail/davmail.log
davmail.carddavReadPhoto=true
davmail.keepDelay=30
davmail.caldavAlarmSound=
```

Any idea?

**Comment by mguessan** [comment]
Looks like you are trying DavMail with an outlook.com address, not sure this will work.

One issue there seems to be itemid length is 140 instead of 144 with standard O365 mailboxes

**Comment by antonioborondo** [comment]
> Looks like you are trying DavMail with an outlook.com address, not sure this will work.
Yes, I'm using an Outlook.com account.

> One issue there seems to be itemid length is 140 instead of 144 with standard O365 mailboxes
I saw your latest commit 2caed4b0664d28f69d572ad21979c24af0207fd9. I can run some tests if you want. Just let me know.

**Comment by mguessan** [comment]
Yes please go ahead, grab latest trunk build or build it yourself and test again

**Comment by antonioborondo** [comment]
The problem is solved with latest `6.2.1-trunk` version. Thank you!
